app/inpututils.cpp

```diff
@@ -189,6 +189,28 @@ QString InputUtils::formatDistanceInProjectUnit( const double distanceInMeters,
   return QString( "%1 %2" ).arg( QString::number( distance, 'f', precision ), abbreviation );
 }
 
+QString InputUtils::formatAreaInProjectUnit( const double areaInSquareMeters, int precision, Qgis::AreaUnit destUnit )
+{
+  Qgis::AreaUnit areaUnit = destUnit;
+
+  if ( areaUnit == Qgis::AreaUnit::Unknown )
+  {
+    areaUnit = QgsProject::instance()->areaUnits();
+  }
+
+  if ( areaUnit == Qgis::AreaUnit::Unknown )
+  {
+    return QString::number( areaInSquareMeters, 'f', precision );
```

Comment:
This should return units similarly to last return in the function, not just a number